src/signals/settings/users/Detail/components/UserForm/index.js

```diff
@@ -40,75 +40,141 @@ const statusOptions = [
 
 const DEFAULT_STATUS_OPTION = 'true';
 
-const UserForm = ({ data, onCancel, onSubmitForm, readOnly }) => {
+const reducer = (state, { field, value }) => ({ ...state, [field]: value });
+
+const UserForm = ({ data, onCancel, onSubmit, readOnly }) => {
   const departments = useSelector(makeSelectDepartments);
-  const departmentList = departments.list.map(({ id, name }) => ({ id, value: name }));
+  const departmentList = departments.list.map(({ id, name }) => ({ id: String(id), value: name }));
+
   const userDepartments = data?.profile?.departments
-    .map(department => departmentList.find(({ value }) => value === department))
+    ?.map(department => departmentList.find(({ value }) => value === department))
     .filter(Boolean) || [];
 
+  const [state, dispatch] = React.useReducer(reducer, {
+    username: data.username,
+    first_name: data.first_name,
+    last_name: data.last_name,
+    note: data.profile && data.profile.note,
+    is_active: data.is_active === undefined ? DEFAULT_STATUS_OPTION : `${data.is_active}`,
+    departments: userDepartments,
+  });
+
+  const onChangeEvent = useCallback(event => {
+    onChange(event.target.name, event.target.value);
+  }, [onChange]);
+
+  const onChange = useCallback((field, value) => { dispatch({ field, value }); }, [dispatch]);
```

Comment:
Shouldn't the format of this function be
```jsx
const onChange = useCallback(
  (field, value) => {
    dispatch({ field, value });
  },
  [dispatch]
);
```

Reply:
It's still less then 100 characters an readable to me, I will split the lines..